lib/src/widgets/board_editor.dart

```diff
@@ -0,0 +1,188 @@
+import 'package:flutter/widgets.dart';
+
+import 'piece.dart';
+import 'positioned_square.dart';
+import '../models.dart';
+import '../board_editor_settings.dart';
+
+/// A chessboard widget where pieces can be dragged around freely (including dragging piece off and onto the board).
+///
+/// This widget can be used as the basis for a fully fledged board editor, similar to https://lichess.org/editor.
+class BoardEditor extends StatefulWidget {
+  const BoardEditor({
+    super.key,
+    required this.size,
+    required this.orientation,
+    required this.pieces,
+    this.settings = const BoardEditorSettings(),
+    this.onTappedSquare,
+    this.onDroppedPiece,
+    this.onDiscardedPiece,
+  });
+
+  /// Visual size of the board.
+  final double size;
+
+  double get squareSize => size / 8;
+
+  /// The pieces to display on the board.
```

Comment:
Could be useful to add here that the `Map` will only be read and never mutated by the `BoardEditor`.